server/src/main/java/org/opensearch/common/util/concurrent/ThreadContext.java

```diff
@@ -215,6 +217,10 @@ public Writeable captureAsWriteable() {
      * if it can't find the task in memory.
      */
     public StoredContext stashWithOrigin(String origin) {
+        SecurityManager sm = System.getSecurityManager();
+        if (sm != null) {
```

Comment:
As we discussed (https://github.com/opensearch-project/OpenSearch/pull/15016#issuecomment-2259158545), we need to make this change backward compatible